pyomo/common/config.py

```diff
@@ -562,10 +589,15 @@ class PathList(Path):
     """
 
     def __call__(self, data):
-        if hasattr(data, "__iter__") and not isinstance(data, str):
-            return [super(PathList, self).__call__(i) for i in data]
-        else:
-            return [super(PathList, self).__call__(data)]
+        try:
+            pathlist = [super(PathList, self).__call__(data)]
+        except TypeError as err:
+            is_not_path_like = "expected str, bytes or os.PathLike" in str(err)
+            if is_not_path_like and hasattr(data, "__iter__"):
+                pathlist = [super(PathList, self).__call__(i) for i in data]
+            else:
+                raise
+        return pathlist
```

Comment:
What is the motivation for this change (especially because I don't see [an obvious] new test that would pass here and fail before)?  I am concerned about putting in a try-except in the validator that is expected to fail in normal cases.

Reply:
In lieu of the changes, I am thinking about changing `PathList.__call__` to something like:

```python
def __call__(self, data):
    is_path_like = isinstance(data, (str, bytes)) or hasattr(data, "__fspath__")
    if is_path_like:
        return [super(PathList, self).__call__(data)]
    else:
        return [super(PathList, self).__call__(i) for i in data]
```

Thoughts?

Reply:
(we were both concurrently  thinking along the exact same lines)